### ISSUE TITLE
Parsing error '=>' expected when parsing typed snippets

### ISSUE THREAD

**Issue by bradyisom** [sveltejs/eslint-plugin-svelte#872]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.12.0

### What version of `eslint-plugin-svelte` are you using?

2.44.1

### What did you do?

```svelte
<script lang="ts">
	function hello() {
		console.log('Hello');
	}
</script>

<h1>Welcome to SvelteKit</h1>
<p>Visit <a href="https://kit.svelte.dev">kit.svelte.dev</a> to read the documentation</p>

{@render editor_button({ onclick: hello })}

<!-- prettier-ignore -->
{#snippet editor_button(foo: {
    onclick: () => void,
})}
  <button on:click={foo.onclick}>Click me</button>
{/snippet}

```

### What did you expect to happen?

I expect no errors

### What actually happened?

Received this error:
```
  12:15  error  Parsing error: '=>' expected
```

### Link to **GitHub Repo** with Minimal Reproducible Example

https://github.com/bradyisom/eslint-typed-snippets

### Additional comments

To repro with GitHub repo, `yarn install` and `yarn lint`

**Comment by bradyisom** [comment]
Perfect! Thank you for the quick response. I can verify that this bug is resolved with the latest version.
